sources/packages/backend/libs/integrations/src/institution-integration/ece-integration/ece-response.processing.service.ts

```diff
@@ -18,38 +45,285 @@ export class ECEResponseProcessingService {
   constructor(
     config: ConfigService,
     private readonly integrationService: ECEResponseIntegrationService,
+    private readonly institutionLocationService: InstitutionLocationService,
+    private readonly confirmationOfEnrollmentService: ConfirmationOfEnrollmentService,
+    private readonly systemUsersService: SystemUsersService,
   ) {
     this.institutionIntegrationConfig = config.institutionIntegration;
   }
 
   /**
    * Process all the available ECE response files in SFTP location.
    * Once the file is processed, it gets deleted.
-   * @returns Process summary.
+   * @returns Process summary result.
    */
   async process(): Promise<ProcessSummaryResult[]> {
-    // Get the list of all files from SFTP ordered by file name.
-    const fileSearch = new RegExp(
-      `^\\/\\w{4}\\/${ECE_RESPONSE_FILE_NAME}$`,
-      "i",
+    // Get all the institution codes who are enabled for integration.
+    const integrationEnabledInstitutions =
+      await this.institutionLocationService.getAllIntegrationEnabledInstitutionCodes();
+    const filePaths = integrationEnabledInstitutions.map((institutionCode) =>
+      path.join(
+        this.institutionIntegrationConfig.ftpResponseFolder,
+        institutionCode,
+        ECE_RESPONSE_FILE_NAME,
+      ),
     );
-    const filePaths = await this.integrationService.getResponseFilesFullPath(
-      this.institutionIntegrationConfig.ftpResponseFolder,
-      fileSearch,
+    // Process all the files in parallel.
+    const result = await processInParallel<ProcessSummaryResult, string>(
+      (remoteFilePath: string) =>
+        this.processDisbursementsInECEResponseFile(remoteFilePath),
+      filePaths,
     );
-    const result: ProcessSummaryResult[] = [];
-    // TODO: Processing of ECE files to be implemented.
-    // This code will be updated.
-    for (const filePath of filePaths) {
-      result.push({
-        summary: [`Reading file ${filePath}`],
-        warnings: [],
-        errors: [],
-      });
-    }
     return result;
   }
 
+  /**
+   * Process individual ECE response file sent by institution.
+   * @param remoteFilePath ECE response file path.
+   * @returns process summary result.
+   */
+  private async processDisbursementsInECEResponseFile(
+    remoteFilePath: string,
+  ): Promise<ProcessSummaryResult> {
+    const processSummary: ProcessSummaryResult = new ProcessSummaryResult();
```

Comment:
I do not see a reason to be explicit about the type here when it is initialized with the `new`.